src/datautils/surfacedata.ts

```diff
@@ -118,6 +64,99 @@ function getSurfaceLines(mappedSurfaces: any, trajectory: number[][]): SurfaceLi
   return lines;
 }
 
+function generateGroupAreas(groups: any, trajectory: any): any {
+  const groupAreas: any = groups.map((g: any, i: number) => {
+    const next: any = i + 1 < groups.length ? groups[i + 1] : null;
+    return {
+      id: g.id,
+      color: convertColor(g.color),
+      data: trajectory.map((p: any, j: any) => [p[0], g.top[j], next ? next.top[j] : null]),
+    };
+  });
+  return groupAreas;
+}
+
+function mapGroups(stratGroups: any, surfaceAreas: any): any {
+  const groups: any = Array.from(stratGroups.values())
+    .sort((a: any, b: any) => a.age - b.age)
+    .filter((g: any) => {
+      const surfaces: string = surfaceAreas[g.name];
+      const isValid = surfaces && surfaces.length > 0;
+      if (!isValid) {
+        console.warn(`Intersection surface group '${g.name}' has no valid entries and will be discarded.`);
+      }
+      return isValid;
+    })
+    .map((g: any, i: number) => {
+      const surface = surfaceAreas[g.name];
+      const top = surface[0];
+      return {
+        id: g.name,
+        color: unassignedColorScale(i),
+        top: top.data.map((d: number[]) => d[1]),
```

Comment:
To make this code a little bit easier to understand, creating lamda functions of the content of the filter and map functions could be a good idea. 
`someArray.sort(onAge).filter(onlyValidGroups).map(toType)` or similar is easy to read.

Also I can see the use of `any` quite a lot in the code. Surly we know the types of most of these variables?!

Reply:
I'm kind of torn on this. I know "Clean code"(as in the book) prefers functions and has a hard limit of max four lines per function to enforce this (you can configure this as a rule in ESLint f.ex.). I find it easier to read this as sequential code rather than alot of small functions which fragments the code, atleast if you are interested in the nitty gritty details rather than getting an overview of what it does. Sometimes I feel like splitting it up into functions are like reading a book where you have to read footnotes for every sentence. But of course there are limits to how much code you can/should have in a function (thought I think 4 lines is too strict) and there is the added benefit of function name documenting intent.

Reply:
As for any, at one point we decided that we provide types for the users of the component, but would not be strict with types internally. In surface data we combine and transform data and have quite a few temporary structures. I'm not sure if defining these types will be worth the effort.

Reply:
Cleaned up use of any

Reply:
Better.
I still think splitting up into smaller functions are better unless understanding every bit of detail is important.
Most of the time it is ok to understand things on a higher level, and then dive into details if needed.
I guess we are discussing one of the main differences between imperative and functional programming, where functional programming is more about what and imperative is more about how. Different ways of writing and reading code.